File: recipes/java-recipes/spring-boot-postgres-recipe/src/main/java/com/trendyol/stove/examples/java/spring/infra/boilerplate/postgres/PostgresConfiguration.java
package com.trendyol.stove.examples.java.spring.infra.boilerplate.postgres;

import io.r2dbc.spi.ConnectionFactory;

import org.springframework.context.annotation.Bean;
import org.springframework.context.annotation.Configuration;
import org.springframework.r2dbc.core.DatabaseClient;

@Configuration
public class PostgresConfiguration {

    @Bean
    public DatabaseClient databaseClient(ConnectionFactory connectionFactory) {
        return DatabaseClient.create(connectionFactory);
    }
}

File: examples/spring-standalone-example/src/main/kotlin/stove/spring/standalone/example/infrastructure/postgres/PostgresConfiguration.kt
package stove.spring.standalone.example.infrastructure.postgres

import io.r2dbc.spi.ConnectionFactory
import org.springframework.context.annotation.Bean
import org.springframework.context.annotation.Configuration
import org.springframework.data.r2dbc.core.R2dbcEntityTemplate
import org.springframework.r2dbc.core.DatabaseClient

@Configuration
class PostgresConfiguration {
  @Bean
  fun r2dbcEntityTemplate(connectionFactory: ConnectionFactory): R2dbcEntityTemplate =
    R2dbcEntityTemplate(connectionFactory)

  @Bean
  fun databaseClient(connectionFactory: ConnectionFactory): DatabaseClient =
    DatabaseClient.create(connectionFactory)
}
